modules/tooling-support/src/main/java/org/mule/runtime/module/tooling/internal/DefaultConnectivityTestingService.java

```diff
@@ -65,21 +66,26 @@ public ConnectionValidationResult testConnection()
                 throw new MuleRuntimeException(e);
             }
         }
+        Object connectivityTestingObject = temporaryArtifact.getMuleContext().getRegistry().get(identifier);
+        if (connectivityTestingObject == null)
+        {
+            throw new ConnectivityTestingObjectNotFoundException(createStaticMessage("It was not possible to find an object to do connectivity testing"));
+        }
         Collection<ConnectivityTestingStrategy> connectivityTestingStrategies = temporaryArtifact.getMuleContext().getRegistry().lookupObjects(ConnectivityTestingStrategy.class);
         for (ConnectivityTestingStrategy connectivityTestingStrategy : connectivityTestingStrategies)
         {
-            if (connectivityTestingStrategy.connectionTestingObjectIsPresent())
+            if (connectivityTestingStrategy.supportsObject(connectivityTestingObject))
```

Comment:
I think that accepts(Object) is a better name than supportsObject(Object), but just an opinion. Keep it as is if you like